pkg/reconciler/autoscaling/kpa/scaler.go

```diff
@@ -166,18 +166,30 @@ func (ks *scaler) handleScaleToZero(pa *pav1alpha1.PodAutoscaler, sks *nv1a1.Ser
 		return scaleUnknown, false
 	} else if pa.Status.IsReady() { // Active=True
 		// Don't scale-to-zero if the PA is active
+		// Unles PA's age is less than StableWindow, which means
+		// that the revision got created but received no traffic
+		// and the enable-fast-scale-down configmap entry was true.
 
 		// Do not scale to 0, but return desiredScale of 0 to mark PA inactive.
 		sw := aresources.StableWindow(pa, config)
+
 		af := pa.Status.ActiveFor()
 		if af >= sw {
 			// We do not need to enqueue PA here, since this will
 			// make SKS reconcile and when it's done, PA will be reconciled again.
 			return desiredScale, false
 		}
+		ks.logger.Debugf("Aggressive scale to 0 at %v, PA Age: %v Window: %v Can? %v Flag: %v",
+			time.Now(), pa.Age(), sw, pa.Age() < sw, config.EnableScaleToZero)
```

Comment:
Take age once before this statement and reuse it throughout, it will slightly differ otherwise.

Reply:
Yeah, this logging was added later... so made it uniform.